app/controllers/concerns/authentication.rb

```diff
@@ -35,14 +41,13 @@ def current_user_signed_in?
     return true if current_user
   end
 
-  def verify_admin
-    redirect_to root_url unless current_user.has_role? :admin
+  def authenticate_user_or_redirect_to_root!
+    if !current_user
+      redirect_to root_url, alert: 'You need to sign in for access to this page.'
```

Comment:
Line is too long. [81/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.